### DISCUSSION TITLE
Flask Login Required Decorator

### DISCUSSION THREAD

**Discussion by nicdz** [rayluo/identity#37] [General]
I like the `@auth.login_required` decorator although its implementation seems to be lacking.

I was able to have the `login_required` decorator redirect the client to the `templates/identity/login.html` page like recommended in your GitHub.  When redirected, the url remains the same as what the client entered. Would be possible to have the redirect URL set to a route of our choice, like `/auth/login`?

I saw that the Abstract Base Class (https://identity-library.readthedocs.io/en/stable/abc.html) has the login and logout built into the redirect URI. Seen in the `redirect_uri` parameter:

> For example, if your input here is https://example.com/x/y/z/redirect, then your project’s redirect page will be mounted at “/x/y/z/redirect”, login page will be at “/x/y/z/login”, and logout page will be at “/x/y/z/logout”.

Is this possible with the flask.auth class?

**Comment by rayluo** [comment]
You can already customize the redirect_uri by using [the `redirect_uri` parameter](https://identity-library.readthedocs.io/en/latest/abc.html#identity.web.WebFrameworkAuth.params.redirect_uri), and the Flask's subclass inherits the same parameter, as documented by the last line of [its `__init__()`'s doc](https://identity-library.readthedocs.io/en/latest/flask.html#identity.flask.Auth.__init__).
